tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -166,61 +167,81 @@ def assert_alert_rule_trigger_migrated(alert_rule_trigger):
 
 
 def build_sentry_app_compare_blob(
-    sentry_app_config: list[dict[str, str]]
-) -> list[dict[str, str | None]]:
+    sentry_app_config: list[dict[str, Any]],
+) -> list[dict[str, Any | None]]:
```

Comment:
nit; `dict[str, Any]` is just a dict - is there something more specific about this? `ComparisonData = dict[str, str | int]` - kind of thing might be nice - better yet, if we know what keys we want to require we could define the dict even more.

By creating these more specific types, we can reduce the number of tests needed to get 100% coverage (for example, the unit tests here should cover every possible type of any to make sure they're handled correctly).

This test is a pretty good example of why giving these types a name is helpful, we import the types and apply them here - so we get type safety and it will stay up to date from the import if we define them more strictly in the future.